packages/machine/src/protocol/install-virtual-app.ts

```diff
@@ -99,38 +114,45 @@ export const INSTALL_VIRTUAL_APP_PROTOCOL: ProtocolExecutionFlow = {
     // M3
     (message: ProtocolMessage, context: Context) => {
       const params2 = message.params as InstallVirtualAppParams;
-      context.outbox[0] = message;
-      context.outbox[0].seq = 3;
-      context.outbox[0].fromAddress = params2.respondingAddress;
-      context.outbox[0].toAddress = params2.intermediaryAddress;
-      context.outbox[0].signature = context.signatures[0]; // s4
-      context.outbox[0].signature2 = context.signatures[1]; // s7
+      context.outbox[0] = {
+        ...message,
+        seq: 3,
+        fromAddress: params2.respondingAddress,
+        toAddress: params2.intermediaryAddress,
+        signature: context.signatures[0], // s4
+        signature2: context.signatures[1] // s7
+      };
     },
 
     // wait for M4
     Opcode.IO_SEND_AND_WAIT
+
+    // TODO: Add appIdentityHash to context
+    // Opcode.STATE_TRANSITION_COMMIT
   ]
 };
 
-function proposeStateTransition3(message: ProtocolMessage, context: Context) {
```

Comment:
I'm confused about this rename, it seems not to have been done at the call sites